### PR TITLE
fix: [EDXOLDMNG-271] 500 error appears if user save the certificate with 2 course modes

### PR DESCRIPTION
**Pull request by NiedielnitsevIvan**
*Description*:
 - change course run relation from 1to1 to foreign key;
 - fix config creation without singnatories;
 - fix test after refactor;

*Reviewers*:
 - @NikolayBorovenskiy 

- [ ] Make sure you are inside the devstack container
- [ ] Run `make test-karma`
- [ ] All tests pass

### CODE REVIEW

File: credentials/apps/api/v2/serializers.py [hunk 1]
```diff
@@ -322,29 +323,35 @@ class SignatoryListField(serializers.ListField):
     Populate signatories images in dict files instead of strings with their filenames.
     """
 
-    def to_internal_value(self, data: str) -> Dict[str, Union[str, InMemoryUploadedFile]]:
+    def to_representation(self, data):
+        return super().to_representation(data.all())
+
+    def to_internal_value(self, data: str) -> List[Optional[Dict[str, Union[str, InMemoryUploadedFile]]]]:
```
Comment (NikolayBorovenskiy) [inline credentials/apps/api/v2/serializers.py:329]: `data: str`?

File: credentials/apps/api/v2/tests/test_views.py [hunk 2]
```diff
@@ -767,6 +767,22 @@ def test_post_for_staff(self):
...
+        self.authenticate_user(self.superuser)
+        data = {
+            "course_id": "course-v1:edX+DemoX+Demo_Course",
+            "certificate_type": "honor",
+            "title": "Name of the certificate",
+            # NOTE: In tests, parsers do not work properly, unlike real queries.
+            # It is in this form that data comes from the studio when creating
+            # a certificate configuration without signatures.
+            # This is the case for this test.
+            "signatories": ["image", "name", "organization", "title"],
```
Comment (NikolayBorovenskiy) [inline credentials/apps/api/v2/tests/test_views.py:780]: be ready to get questions of global review.

File: credentials/apps/api/v2/serializers.py (hunk 1 above)
Comment (NiedielnitsevIvan) [reply credentials/apps/api/v2/serializers.py]: Fixed

File: credentials/apps/api/v2/tests/test_views.py (hunk 2 above)
Comment (NiedielnitsevIvan) [reply credentials/apps/api/v2/tests/test_views.py:780]: Okay, I'll try to describe this problem on the global review.

File: credentials/apps/api/v2/serializers.py [hunk 3]
```diff
@@ -322,29 +323,35 @@ class SignatoryListField(serializers.ListField):
...
         Converts a list of dict that is a string to a list of dict.
         """
-        data_ = self.populate_signatories(data, self.context["request"].FILES)  # pylint: disable=no-member
-        return super().to_internal_value(data_)
+        try:
+            jsonified_data = list(map(ast.literal_eval, copy(data)))
+        except ValueError:
+            return []
+        else:
+            data_ = self.populate_signatories(
```
Comment (NikolayBorovenskiy) [inline credentials/apps/api/v2/serializers.py:338]: Please fix this. Use `_data` instead of `data_`

Comment (NiedielnitsevIvan) [reply credentials/apps/api/v2/serializers.py]: Corrected

**Review by NikolayBorovenskiy — APPROVED** [review]

**Pull request merged**